### ISSUE TITLE
Add filtering capabilities

### ISSUE THREAD

**Issue by kaelad02** [kaelad02/region-active-effects#1]
Instead of applying an effect to every token in a Region, it'd be nice to be able to add a filter to the behaviors to choose which kind of tokens to apply it to.

I'm soliciting feedback here, so if you have a suggestion of what kind of filters would be useful, please add a comment here. Just keep in mind that this module is system-agnostic, so if there is something specific to a game system you use we might have to tweak the requirement accordingly. Even so, I still want to know your ideas.

Making a list:

Core
- actor type

PF2e
- use the system's Predicates to allow any kind of filtering

dnd5e
1. Creature Type (e.g. Beast or Fey)
2. Creature Size (e.g. Tiny or Small)
3. Health (e.g. below 50% HP)
4. Armor Type (e.g. wearing Heavy Armor)

**Comment by Mermut** [comment]
Player characters vs NPC would be the most obvious filter for me
Next would be able to filter on actor 'traits' (specifically the trait section for PF2e, but I'm not sure if other systems have similar actor descriptors that are in a specific field on the actor)

**Comment by kaelad02** [comment]
> Player characters vs NPC would be the most obvious filter for me

@Mermut Agreed, seems obvious. Would testing the actor type be better (e.g. character vs. npc) or the token disposition (e.g. friendly vs. hostile)?

**Comment by Peach774** [comment]
> > Player characters vs NPC would be the most obvious filter for me
> 
> @Mermut Agreed, seems obvious. Would testing the actor type be better (e.g. character vs. npc) or the token disposition (e.g. friendly vs. hostile)?

having both as options would probably be ideal if it could be stacked with other checks, like traits from pf2e

**Comment by Mermut** [comment]
> > > Player characters vs NPC would be the most obvious filter for me
> > 
> > 
> > @Mermut Agreed, seems obvious. Would testing the actor type be better (e.g. character vs. npc) or the token disposition (e.g. friendly vs. hostile)?
> 
> having both as options would probably be ideal if it could be stacked with other checks, like traits from pf2e

I agree, both would ideal, but if I had to choose, I'd do NPC vs PC

**Comment by kaelad02** [comment]
I spoke with some people on the Pathfinder on Foundry Discord server and it sounds like I can use the system's Predicates feature so you can add any kind of filtering you want. The fact it's commonly used elsewhere in the system is a plus.

**Comment by Kharmans** [comment]
A good filter would be **Enemies**, **Allies**, **Hostiles**, etc.
